backend/app/server.py

```diff
@@ -44,7 +48,27 @@ async def ingest_files(
         if thread is None:
             raise HTTPException(status_code=404, detail="Thread not found.")
 
-    return ingest_runnable.batch([file.file for file in files], config)
+    if config["configurable"].get("show_progress_bar"):
+        ingest_runnable.abatch_as_completed([file.file for file in files], config)
+
+        # The return type of the IngestRunnable is not compatible with the
+        # FastAPI StreamingResponse (the data must have an `.encode()`
+        # property). The results are (int, list) tuples, so lets use orjson to
+        # byte the tuple
+        async def decode_ingestion_response(ingest_generator):
+            async for x in ingest_generator:
+                yield orjson.dumps(x)
+                await asyncio.sleep(3)  # Debug: to demonstrate streaming
+
+        return StreamingResponse(
```

Comment:
ideally we'd use a more standard streaming format, eg Server Sent Events

Reply:
I'm not familiar with Server Sent Events, but just for completeness, what's the concern with `fastapi.StreamingResponse()`?